tools/util.wdl

```diff
@@ -417,3 +417,173 @@ task make_coverage_regions_beds {
         maxRetries: max_retries
     }
 }
+
+task global_phred_scores {
+    meta {
+        description: "Calculates statistics about PHRED scores of the input BAM."
+    }
+
+    input {
+        File bam
+        String prefix = basename(bam, ".bam")
+        Int memory_gb = 4
+        Int modify_disk_size_gb = 0
+        Int max_retries = 1
+    }
+
+    Float bam_size = size(bam, "GiB")
+    Int disk_size_gb = ceil(bam_size) + 10 + modify_disk_size_gb
+
+    String outfile_name = prefix + ".global_PHRED_scores.tsv"
+
+    command <<<
+        set -euo pipefail
+
+        BAM="~{bam}" PREFIX="~{prefix}" python3 - <<END
```

Comment:
Just a general comment, should we be inlining python scripts? Is there some limit to complexity/length that should be a cutoff?

Reply:
Good question. Historically, these inline Python scripts (of which there's a few now) were just implemented because it was a fast and easy. Now that I'm proliferating the practice into different places, I think it's a good idea to check whether it's a practice we want to maintain.

The most obvious alternative is to package these scripts with the `util` image. That should be an easy copy+paste without any significant changes to the Docker. Except I believe there's a script that runs on a `gtfparse` image pulled from BioContainers. That one would be trickier to change. IMO we wouldn't want to start maintaining another Docker image just to avoid some inlined Python. We could add `gtfparse` to the util image's dependencies, but then we run the risk of turning that image into a bloated "kitchen sink".

I think it would be a more maintainable and "prettier" solution to package these as scripts with `util`, but it does have a downside. These WDL tasks would become black boxes, unless you go and track down the source Docker directory.

I think the complexity/length angle could be a good way to decide what belongs inline VS packaged in Docker.

I have to chew on this problem some more, so why don't we push this question off to another PR/discuss offline.

TLDR; I don't think inlining Python is _terrible_, but we should avoid it just for "best practice" reasons. The question is what's the best alternative, and I don't have an immediate answer.

Reply:
Agreed that it doesn't need to be resolved here. I'm not sure that I'm opposed to inline Python. I just think we may want to consider some guidance on length/complexity that would guide splitting it out into a standalone script. If the code is general purpose (such as this), I could see having a top-level `scripts` in the repo and the script lives there.